
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="canonical" href="https://pmitev.github.io/to-awk-or-not/Case_studies/manipulating_vcf/">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.2, mkdocs-material-7.2.6">
    
    
      
        <title>Manipulating the output from a genome analysis - vcf and gff - To awk or not</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.802231af.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.3f5d1f46.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
    
      


    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <script>function __prefix(e){return new URL("../..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
      <script>var palette=__get("__palette");if(null!==palette&&"object"==typeof palette.color)for(var key in palette.color)document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#manipulating-the-output-from-a-genome-analysis-vcf-and-gff" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="To awk or not" class="md-header__button md-logo" aria-label="To awk or not" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            To awk or not
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Manipulating the output from a genome analysis - vcf and gff
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M7 10a2 2 0 0 1 2 2 2 2 0 0 1-2 2 2 2 0 0 1-2-2 2 2 0 0 1 2-2m10-3a5 5 0 0 1 5 5 5 5 0 0 1-5 5H7a5 5 0 0 1-5-5 5 5 0 0 1 5-5h10M7 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3h10a3 3 0 0 0 3-3 3 3 0 0 0-3-3H7z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/pmitev/to-awk-or-not/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="To awk or not" class="md-nav__button md-logo" aria-label="To awk or not" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    To awk or not
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/pmitev/to-awk-or-not/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../1.Simple_example/" class="md-nav__link">
        1.Simple examples
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../2.Teasing_with_grep/" class="md-nav__link">
        2.Teasing with grep
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../3.Shell_we_awk/" class="md-nav__link">
        3.Shell we awk
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../4.Brief_commands/" class="md-nav__link">
        4.Brief commands
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../5.String_manipulation/" class="md-nav__link">
        5.String manipulation
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../6.One_line_programs/" class="md-nav__link">
        6.One line programs
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Python_vs_awk/" class="md-nav__link">
        Python vs. awk
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_9" type="checkbox" id="__nav_9" >
      
      <label class="md-nav__link" for="__nav_9">
        Exercises
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Exercises" data-md-level="1">
        <label class="md-nav__title" for="__nav_9">
          <span class="md-nav__icon md-icon"></span>
          Exercises
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/Exercises/" class="md-nav__link">
        Warming up
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/01.Simple_arithmetic/" class="md-nav__link">
        01.Simple arithmetic *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/02.Data_extraction/" class="md-nav__link">
        02.Data extraction *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/03.Data_extraction/" class="md-nav__link">
        03.Data extraction **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/04.Data_manipulation/" class="md-nav__link">
        04.Data manipulation **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/05.Easy_tricks/" class="md-nav__link">
        05.Easy tricks *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/06.Bioinformaticians_corner/" class="md-nav__link">
        06.Bioinformaticians corner *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/Advanced_data_analysis/" class="md-nav__link">
        Advanced data analysis ****
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/Carpool/" class="md-nav__link">
        Carpool ***
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/Difficult_data/" class="md-nav__link">
        Difficult to extract data ****
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/Scrubbing/" class="md-nav__link">
        Scrubbing a web page **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/String_manipulation/" class="md-nav__link">
        String manipulation **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/awk-json/" class="md-nav__link">
        Awk and json **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../Exercises/Linear_regression/" class="md-nav__link">
        Simple linear regresion ***
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../List/" class="md-nav__link">
        = Case Studies =
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_11" type="checkbox" id="__nav_11" >
      
      <label class="md-nav__link" for="__nav_11">
        More with awk
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="More with awk" data-md-level="1">
        <label class="md-nav__title" for="__nav_11">
          <span class="md-nav__icon md-icon"></span>
          More with awk
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/User_defined_functions/" class="md-nav__link">
        User defined functions
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/Command_params/" class="md-nav__link">
        Parameters on the command line
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/Input_output/" class="md-nav__link">
        Input/Output to external program
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/Running_average/" class="md-nav__link">
        Running averages
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Fixed_size_fields/" class="md-nav__link">
        Fixed size fields
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Backreferences/" class="md-nav__link">
        Backreferences
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="https://www.gnu.org/software/gawk/manual/html_node/Multiple-Line.html" class="md-nav__link">
        Multi-Line records
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../arrays/" class="md-nav__link">
        Awk arrays
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../awk_bash/" class="md-nav__link">
        Awk or Bash
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Localization/" class="md-nav__link">
        Localization problems
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="http://www.thegeekstuff.com/2011/06/awk-nawk-gawk/" class="md-nav__link">
        Awk vs. nawk vs. gawk
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Binder/" class="md-nav__link">
        GitHub & Binder
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../handy/" class="md-nav__link">
        Handy to have
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../tinyutils/" class="md-nav__link">
        tinyutils
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Links/" class="md-nav__link">
        Links
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../About/" class="md-nav__link">
        About
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#getting-the-raw-files" class="md-nav__link">
    Getting the RAW files
  </a>
  
    <nav class="md-nav" aria-label="Getting the RAW files">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-annotation-of-the-fly-genome" class="md-nav__link">
    The annotation of the fly genome:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-genome-sequence-used-in-this-project" class="md-nav__link">
    The genome sequence used in this project:
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-comparison-of-all-the-variants-in-the-cell-lines-in-the-freeze-project" class="md-nav__link">
    The comparison of all the variants in the cell lines in the freeze project
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#starting-to-work-with-the-data" class="md-nav__link">
    Starting to work with the data
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#what-do-we-need-to-get" class="md-nav__link">
    What do we need to get
  </a>
  
    <nav class="md-nav" aria-label="What do we need to get">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#tips-before-starting" class="md-nav__link">
    Tips before starting
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#the-exercise" class="md-nav__link">
    The exercise
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#chromosome-4" class="md-nav__link">
    Chromosome 4
  </a>
  
    <nav class="md-nav" aria-label="Chromosome 4">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#follow-up-task" class="md-nav__link">
    Follow-up task:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#snps-and-indels" class="md-nav__link">
    SNPs and INDELs
  </a>
  
    <nav class="md-nav" aria-label="SNPs and INDELs">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#follow-up-task_1" class="md-nav__link">
    Follow-up task:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genes-with-variants" class="md-nav__link">
    Genes with variants
  </a>
  
    <nav class="md-nav" aria-label="Genes with variants">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#follow-up-task_2" class="md-nav__link">
    Follow-up task:
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
        
          <li class="md-nav__item">
  <a href="#genescdss-only" class="md-nav__link">
    Genes/CDSs only
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#final-list-of-variants" class="md-nav__link">
    Final list of variants
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#final-result" class="md-nav__link">
    Final result
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#extra-task" class="md-nav__link">
    Extra task
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/pmitev/to-awk-or-not/edit/master/docs/Case_studies/manipulating_vcf.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="manipulating-the-output-from-a-genome-analysis-vcf-and-gff">Manipulating the output from a genome analysis - vcf and gff</h1>
<blockquote>
<p>Problem formulated an presented at the workshop by <a href="https://katalog.uu.se/profile/?id=N2-1277"><strong>Jonas Söderberg</strong></a>,<br />
Department of Cell and Molecular Biology, <em>Molecular Evolution</em></p>
</blockquote>
<p>We have a comparison between a number of different fly cell lines. These are found in a huge vcf file (dgrp2.vcf). We also have the annotations in a gff3 file (Drosophila_melanogaster.BDGP6.28.101.chr.gff3) and the genome itself in a fasta file.</p>
<h2 id="getting-the-raw-files">Getting the RAW files</h2>
<h3 id="the-annotation-of-the-fly-genome">The annotation of the fly genome:</h3>
<p>Find the GFF3 <a href="https://tinyurl.com/y2opo93p" title="Drosophila_melanogaster.BDGP6.28.101.chr.gff3">here</a>.</p>
<h3 id="the-genome-sequence-used-in-this-project">The genome sequence used in this project:</h3>
<p>The genome is located <a href="https://tinyurl.com/yyldprwp" title="Drosophila_melanogaster.BDGP6.28.dna.toplevel.fa">here</a>.</p>
<h3 id="the-comparison-of-all-the-variants-in-the-cell-lines-in-the-freeze-project">The comparison of all the variants in the cell lines in the freeze project</h3>
<p>Finally, <a href="http://dgrp2.gnets.ncsu.edu/data.html" title="dgrp2.vcf">here</a> is the VCF file.</p>
<h3 id="starting-to-work-with-the-data">Starting to work with the data</h3>
<p>Let's say we want to find out all genes that contains a variant and all variants that are located within a gene. What do we want to do first? Take a look at the vcf file. That is the one that contains all the variants. Then look at the gff file, which contains the genes and other annotations. Finally, take a look at the DNA sequence. You will need to combine all three to answer the question. Also, to make this faster, lets just look at <strong>chromosome 4</strong>, which means we have to extract that data as well.</p>
<h3 id="what-do-we-need-to-get">What do we need to get</h3>
<ul>
<li>Fles containing only chromosome 4</li>
<li>Positions for SNPs and INDELs</li>
<li>Positions for genes and CDSs</li>
<li>Separation of variants (SNPs and INDELs) into two groups, inside and outside genes (and CDSs)</li>
<li>Separation of genes/CDSs into those with and without variants (and maybe how many there are per gene)</li>
</ul>
<h4 id="tips-before-starting">Tips before starting</h4>
<ul>
<li>For readability of the vcf file:</li>
</ul>
<p><code>awk '{print $1"\t"$2"\t"$3"\t"$4"\t"$5"\t"$6"\t"$7"\t"$8"\t"$9}' dgrp2.vcf &gt; dgrp2_trimmed.vcf &amp;</code></p>
<ul>
<li>If your awk takes too long to run (10h or so) - Make sure you have the latest awk, and also you might want to think about parallelisation.</li>
<li>My solutions also work for files with more than one chromosome, so in some cases they are longer than needed for your exercise. I left it so on purpose.</li>
<li><em>SNPs</em> and <em>INDELs</em> are collectively named <strong>variants</strong></li>
<li><em>Genes</em> and <em>CoDing Sequences (CDSs)</em> are sometimes just collectively called <strong>genes</strong></li>
</ul>
<h3 id="the-exercise">The exercise</h3>
<p>Identify the steps you need to do and what each step does. Open the hints if you get stuck.</p>
<h3 id="chromosome-4">Chromosome 4</h3>
<details><summary><em>Hint</em>, what do we need?</summary><p>Extract chr4 from the vcf and the gff and make new files</p>
</details>
<details><summary><em>Hint</em></summary><p>All lines from chromosome 4 start with a <em>4</em></p>
</details>
<details><summary><em>Solution</em></summary><p><code>awk '/^4/{print $0}' Drosophila_melanogaster.BDGP6.28.101.chr.gff3 &gt; Drosophila_melanogaster.chr4.gff3</code></p>
<p><code>awk '/^4/{print $0}' dgrp2_trimmed.vcf &gt; dgrp2_chr4.vcf</code></p>
</details>
<h4 id="follow-up-task"><em>Follow-up task:</em></h4>
<p>Count and sort the different genomic features in chromosome 4 by number.</p>
<details><summary><strong>Task result example</strong></summary><div class="highlight"><pre><span></span><code>   1 chromosome
   1 snoRNA
   2 pre_miRNA
   7 pseudogene
  11 pseudogenic_transcript
  26 ncRNA_gene
  31 ncRNA
  79 gene
 295 mRNA
 338 three_prime_UTR
 571 five_prime_UTR
2740 CDS
3155 exon
</code></pre></div>
</details>
<h3 id="snps-and-indels">SNPs and INDELs</h3>
<details><summary><em>Hint</em>, which output do we want?</summary><p>Get distribution of variants and list them in two separate files. For a bonus plot of the lengths of the INDELS, get the length of all INDELS into a third file</p>
</details>
<details><summary><em>Hint</em></summary><p>Remove lines beginning with # (grep)</p>
</details>
<details><summary><em>Hint</em></summary><p>If columns 4 and 5 have different length, it's an INDEL. Otherwise it's a SNP.</p>
</details>
<details><summary><em>Hint</em></summary><p>You want the output to be a file with columns 1, 2, 4 and 5, a classifier (SNP or INDEL) and finally the length of the INDEL (put "-" for the SNPs)</p>
</details>
<details><summary><em>Solution</em></summary><p><code>cat dgrp2_chr4.vcf | grep -v "#" | awk '{if (length($4)&gt;1||length($5)&gt;1){a="INDEL";b=length($4)-length($5);cnt[b]+=1;} else {a="SNP";b="-";} printf("%s\t%s\t%s\t%s\t%s\t%s\n", $1, $2, a, b, $4, $5) &gt; "indels_Drosophila_chr4";}END{for (x in cnt){print x,cnt[x] &gt; "distr_Drosophila_chr4"}}'</code></p>
</details>
<h4 id="follow-up-task_1"><em>Follow-up task:</em></h4>
<p>Print nucleotide substitution that these SNPs introduce sorted by number. Remember the coins...</p>
<details><summary><em>Task result example</em></summary><div class="highlight"><pre><span></span><code>1182 C-&gt;T  
1133 G-&gt;A  
 932 A-&gt;G  
 929 A-&gt;T  
 892 T-&gt;A  
 880 T-&gt;C  
 639 G-&gt;T  
 621 C-&gt;A  
 436 A-&gt;C  
 396 T-&gt;G  
 372 G-&gt;C  
 357 C-&gt;G
</code></pre></div>
</details>
<h3 id="genes-with-variants">Genes with variants</h3>
<details><summary><em>Hint</em>, how do we get those?</summary><p>Compare back and separate the annotation into features that do and don’t have variants. For a bonus, also record the number of variants in each feature</p>
</details>
<details><summary><em>Hint</em></summary><p>Make an index using the previous output to identify positions of variants</p>
</details>
<details><summary><em>Hint</em></summary><p>For each feature in the gff, check all position it covers to see if they are in your index, if so print to one file. If not, print to another.</p>
</details>
<details><summary><em>Solution</em></summary><p><code>awk 'FNR==NR{a[$1,$2]="T"; next}{ hits=0; for(N=$4; N&lt;=$5; N++) { if (a[$1,N] == "T") {hits+=1}} if (hits&gt;0) {print hits "\t" $0 &gt; "haveSNPINDEL_Drosophila_chr4.gff"} else {print $0 &gt; "noSNPINDEL_Drosophila_chr4.gff"}}' indels_Drosophila_chr4 Drosophila_melanogaster.chr4.gff3</code></p>
</details>
<h4 id="follow-up-task_2"><em>Follow-up task:</em></h4>
<p>Count and sort the SNPs (not INDELs) in your output and compare to the output from the first step.</p>
<details><summary><em>Task result example</em></summary><div class="highlight"><pre><span></span><code>   1 chromosome
   1 pre_miRNA
   1 snoRNA
   6 pseudogene
   9 pseudogenic_transcript
  22 ncRNA_gene
  28 ncRNA
  79 gene
 264 three_prime_UTR
 290 five_prime_UTR
 295 mRNA
1798 CDS
2181 exon
</code></pre></div>
</details>
<h3 id="genescdss-only">Genes/CDSs only</h3>
<details><summary><em>Hint</em>, what features do we look for?</summary><p>Filter for genes and CDSs before doing the analysis.</p>
</details>
<details><summary><em>Hint</em></summary><p>Only genes and CDSs are interesting to us. Make a gff without the rest of the features.</p>
</details>
<details><summary><em>Solution</em></summary><p><code>awk '{if ($3=="gene" || $3=="CDS") print $0}' Drosophila_melanogaster.chr4.gff3 &gt; Drosophila_melanogaster.chr4_genesCDSs.gff3</code></p>
</details>
<h3 id="final-list-of-variants">Final list of variants</h3>
<details><summary><em>Hint</em>, how do we classify the variants?</summary><p>Repeat step 3 for the SNPs/INDELs themselves, to see which are actually located inside genes</p>
</details>
<details><summary><em>Hint</em></summary><p>Make an index of all genes/CDSs (from your gff), where start and stop are paired</p>
</details>
<details><summary><em>Hint</em></summary><p>For each feature from your step 2 file, check the position agains the index and print whether or not the variant is inside a gene.</p>
</details>
<details><summary><em>Solution</em></summary><p><code>awk 'FNR==NR{ingene[$1,$4]=$5; next}{state="Not in gene";for (pair in ingene) {split(pair, t, SUBSEP); if ($1==t[1] &amp;&amp; $2&gt;=t[2] &amp;&amp; $2&lt;=ingene[t[1],t[2]]) {state=(t[1] " " t[2] " " ingene[t[1],t[2]])}} print $0, " ", state }' Drosophila_melanogaster.chr4_genesCDSs.gff3 indels_Drosophila_chr4 &gt; SNPsInGenes_Drosophila_ch4</code></p>
</details>
<h3 id="final-result">Final result</h3>
<p>Here we can see all SNPs and INDELs which are inside a relevant region. We have successfully made two gff:s containing all gene positions for genes with variants and genes without. Along the way we also got a list of all genes that contain variants too.</p>
<h3 id="extra-task">Extra task</h3>
<p>Re-do the files but this time include the gene ID (i.e. FBtr0089178 from column nine) and translate that into the full gene name found in this <a href="http://ftp.flybase.org/releases/FB2020_06/precomputed_files/genes/fbgn_fbtr_fbpp_expanded_fb_2020_06.tsv.gz">file</a>.</p>
<details><summary><strong>Solution</strong></summary><p><code>awk 'FNR==1{++fileidx} fileidx==1{split($9,a,";|:");ingene[$1,$4,$5]=a[2]; next} fileidx==2{FS="\t";name[$3]=$5} fileidx==3{state="Not in gene";for (trip in ingene) {split(trip, t, SUBSEP); if ($1==t[1] &amp;&amp; $2&gt;=t[2] &amp;&amp; $2&lt;=t[3]) {state=(t[1] "\t" t[2] "\t" t[3] "\t" name[ingene[t[1],t[2],t[3]]])}} print $0, "\t", state }' Drosophila_melanogaster.chr4_genesCDSs.gff3 fbgn_fbtr_fbpp_expanded_fb_2020_06.tsv indels_Drosophila_chr4 &gt; SNPsInNamedGenes_Drosophila_ch4</code></p>
<p>Look at the distribution of genes:</p>
<p><code>awk -F"\t" '{print $10}' SNPsInNamedGenes_Drosophila_ch4 | sort | uniq -c | sort -n</code></p>
</details>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      <a href="https://uppmax.uu.se/" target="_blank" rel="noopener"><img align="right" alt="Uppsala" src="https://pmitev.github.io/to-awk-or-not/images/Uppsala.png" /> </a>
      <a href="https://snic.se/" target="_blank" rel="noopener"><img align="right" alt="SNIC" src="https://pmitev.github.io/to-awk-or-not/images/SNIC.png" /> </a>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.409db549.min.js", "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.756773cc.min.js"></script>
      
        <script src="../../javascripts/config.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>