
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      
        <link rel="canonical" href="https://pmitev.github.io/to-awk-or-not/Exercises/gtf-teaser/">
      
      <link rel="icon" href="../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.2.2, mkdocs-material-7.2.6">
    
    
      
        <title>How to Analyze a Transcriptome Like a Pro - To awk or not</title>
      
    
    
      <link rel="stylesheet" href="../../assets/stylesheets/main.802231af.min.css">
      
        
        <link rel="stylesheet" href="../../assets/stylesheets/palette.3f5d1f46.min.css">
        
      
    
    
    
      
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,400,400i,700%7CRoboto+Mono&display=fallback">
        <style>:root{--md-text-font-family:"Roboto";--md-code-font-family:"Roboto Mono"}</style>
      
    
    
    
    
      


    
    
  </head>
  
  
    
    
      
    
    
    
    
    <body dir="ltr" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo">
  
    
    <script>function __prefix(e){return new URL("../..",location).pathname+"."+e}function __get(e,t=localStorage){return JSON.parse(t.getItem(__prefix(e)))}</script>
    
      <script>var palette=__get("__palette");if(null!==palette&&"object"==typeof palette.color)for(var key in palette.color)document.body.setAttribute("data-md-color-"+key,palette.color[key])</script>
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#how-to-analyze-a-transcriptome-like-a-pro" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
      <header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../.." title="To awk or not" class="md-header__button md-logo" aria-label="To awk or not" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            To awk or not
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              How to Analyze a Transcriptome Like a Pro
            
          </span>
        </div>
      </div>
    </div>
    
      <form class="md-header__option" data-md-component="palette">
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: light)" data-md-color-scheme="default" data-md-color-primary="indigo" data-md-color-accent="indigo"  aria-label="Switch to dark mode"  type="radio" name="__palette" id="__palette_1">
          
            <label class="md-header__button md-icon" title="Switch to dark mode" for="__palette_2" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M7 10a2 2 0 0 1 2 2 2 2 0 0 1-2 2 2 2 0 0 1-2-2 2 2 0 0 1 2-2m10-3a5 5 0 0 1 5 5 5 5 0 0 1-5 5H7a5 5 0 0 1-5-5 5 5 0 0 1 5-5h10M7 9a3 3 0 0 0-3 3 3 3 0 0 0 3 3h10a3 3 0 0 0 3-3 3 3 0 0 0-3-3H7z"/></svg>
            </label>
          
        
          
          
          <input class="md-option" data-md-color-media="(prefers-color-scheme: dark)" data-md-color-scheme="slate" data-md-color-primary="blue" data-md-color-accent="blue"  aria-label="Switch to light mode"  type="radio" name="__palette" id="__palette_2">
          
            <label class="md-header__button md-icon" title="Switch to light mode" for="__palette_1" hidden>
              <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M17 7H7a5 5 0 0 0-5 5 5 5 0 0 0 5 5h10a5 5 0 0 0 5-5 5 5 0 0 0-5-5m0 8a3 3 0 0 1-3-3 3 3 0 0 1 3-3 3 3 0 0 1 3 3 3 3 0 0 1-3 3z"/></svg>
            </label>
          
        
      </form>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
      </label>
      
<div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        
<a href="https://github.com/pmitev/to-awk-or-not/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
      </div>
    
  </nav>
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../.." title="To awk or not" class="md-nav__button md-logo" aria-label="To awk or not" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54z"/></svg>

    </a>
    To awk or not
  </label>
  
    <div class="md-nav__source">
      
<a href="https://github.com/pmitev/to-awk-or-not/" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    GitHub
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../1.Simple_example/" class="md-nav__link">
        1.Simple examples
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../2.Teasing_with_grep/" class="md-nav__link">
        2.Teasing with grep
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../3.Shell_we_awk/" class="md-nav__link">
        3.Shell we awk
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../4.Brief_commands/" class="md-nav__link">
        4.Brief commands
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../5.String_manipulation/" class="md-nav__link">
        5.String manipulation
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../6.One_line_programs/" class="md-nav__link">
        6.One line programs
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Python_vs_awk/" class="md-nav__link">
        Python vs. awk
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_9" type="checkbox" id="__nav_9" >
      
      <label class="md-nav__link" for="__nav_9">
        Exercises
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="Exercises" data-md-level="1">
        <label class="md-nav__title" for="__nav_9">
          <span class="md-nav__icon md-icon"></span>
          Exercises
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Exercises/" class="md-nav__link">
        Warming up
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../01.Simple_arithmetic/" class="md-nav__link">
        01.Simple arithmetic *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../02.Data_extraction/" class="md-nav__link">
        02.Data extraction *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../03.Data_extraction/" class="md-nav__link">
        03.Data extraction **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../04.Data_manipulation/" class="md-nav__link">
        04.Data manipulation **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../05.Easy_tricks/" class="md-nav__link">
        05.Easy tricks *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../06.Bioinformaticians_corner/" class="md-nav__link">
        06.Bioinformaticians corner *
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Advanced_data_analysis/" class="md-nav__link">
        Advanced data analysis ****
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Carpool/" class="md-nav__link">
        Carpool ***
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Difficult_data/" class="md-nav__link">
        Difficult to extract data ****
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Scrubbing/" class="md-nav__link">
        Scrubbing a web page **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../String_manipulation/" class="md-nav__link">
        String manipulation **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../awk-json/" class="md-nav__link">
        Awk and json **
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../Linear_regression/" class="md-nav__link">
        Simple linear regresion ***
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Case_studies/List/" class="md-nav__link">
        = Case Studies =
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_11" type="checkbox" id="__nav_11" >
      
      <label class="md-nav__link" for="__nav_11">
        More with awk
        <span class="md-nav__icon md-icon"></span>
      </label>
      <nav class="md-nav" aria-label="More with awk" data-md-level="1">
        <label class="md-nav__title" for="__nav_11">
          <span class="md-nav__icon md-icon"></span>
          More with awk
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/User_defined_functions/" class="md-nav__link">
        User defined functions
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/Command_params/" class="md-nav__link">
        Parameters on the command line
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/Input_output/" class="md-nav__link">
        Input/Output to external program
      </a>
    </li>
  

          
            
  
  
  
    <li class="md-nav__item">
      <a href="../../More_awk/Running_average/" class="md-nav__link">
        Running averages
      </a>
    </li>
  

          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Fixed_size_fields/" class="md-nav__link">
        Fixed size fields
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Backreferences/" class="md-nav__link">
        Backreferences
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="https://www.gnu.org/software/gawk/manual/html_node/Multiple-Line.html" class="md-nav__link">
        Multi-Line records
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../arrays/" class="md-nav__link">
        Awk arrays
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../awk_bash/" class="md-nav__link">
        Awk or Bash
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Localization/" class="md-nav__link">
        Localization problems
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="http://www.thegeekstuff.com/2011/06/awk-nawk-gawk/" class="md-nav__link">
        Awk vs. nawk vs. gawk
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Other/Binder/" class="md-nav__link">
        GitHub & Binder
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../handy/" class="md-nav__link">
        Handy to have
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../tinyutils/" class="md-nav__link">
        tinyutils
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../Links/" class="md-nav__link">
        Links
      </a>
    </li>
  

    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../About/" class="md-nav__link">
        About
      </a>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    
<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
                  <a href="https://github.com/pmitev/to-awk-or-not/edit/master/docs/Exercises/gtf-teaser.md" title="Edit this page" class="md-content__button md-icon">
                    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25z"/></svg>
                  </a>
                
                
                <h1 id="how-to-analyze-a-transcriptome-like-a-pro">How to Analyze a Transcriptome Like a Pro</h1>
<p>This is just a teaser for the full tutorial:<br />
<a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1/">AWK GTF! How to Analyze a Transcriptome Like a Pro</a></p>
<p>Let's use small file to exercise a bit with the content.</p>
<div class="highlight"><pre><span></span><code>$ wget https://raw.github.com/nachocab/nachocab.github.io/master/assets/transcriptome.gtf
</code></pre></div>
<blockquote>
<p>Hint: to get the line unwrapped in the terminal pipe the output to <code>less -S</code></p>
</blockquote>
<div class="highlight"><pre><span></span><code>$ head transcriptome.gtf <span class="p">|</span> less -S
<span class="c1">##description: evidence-based annotation of the human genome (GRCh37), version 18 (Ensembl 73)</span>
<span class="c1">##provider: GENCODE</span>
<span class="c1">##contact: gencode@sanger.ac.uk</span>
<span class="c1">##format: gtf</span>
<span class="c1">##date: 2013-09-02</span>
chr1    HAVANA  exon    <span class="m">173753</span>  <span class="m">173862</span>  .   -   .   gene_id <span class="s2">&quot;ENSG00000241860.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENST00000466557.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;processed_transcript&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;RP11-34P13.13&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;RP11-34P13.13-001&quot;</span><span class="p">;</span> exon_number <span class="m">1</span><span class="p">;</span>  exon_id <span class="s2">&quot;ENSE00001947154.2&quot;</span><span class="p">;</span>  level <span class="m">2</span><span class="p">;</span> tag <span class="s2">&quot;not_best_in_genome_evidence&quot;</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000002480.3&quot;</span><span class="p">;</span> havana_transcript <span class="s2">&quot;OTTHUMT00000007037.2&quot;</span><span class="p">;</span>
chr1    HAVANA  transcript  <span class="m">1246986</span> <span class="m">1250550</span> .   -   .   gene_id <span class="s2">&quot;ENSG00000127054.14&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENST00000478641.1&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;CPSF3L&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;retained_intron&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;CPSF3L-006&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000003330.11&quot;</span><span class="p">;</span> havana_transcript <span class="s2">&quot;OTTHUMT00000009365.1&quot;</span><span class="p">;</span>
chr1    HAVANA  CDS <span class="m">1461841</span> <span class="m">1461911</span> .   +   <span class="m">0</span>   gene_id <span class="s2">&quot;ENSG00000197785.9&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENST00000378755.5&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;ATAD3A&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;ATAD3A-003&quot;</span><span class="p">;</span> exon_number <span class="m">13</span><span class="p">;</span>  exon_id <span class="s2">&quot;ENSE00001664426.1&quot;</span><span class="p">;</span>  level <span class="m">2</span><span class="p">;</span> tag <span class="s2">&quot;basic&quot;</span><span class="p">;</span> tag <span class="s2">&quot;CCDS&quot;</span><span class="p">;</span> ccdsid <span class="s2">&quot;CCDS31.1&quot;</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000575.6&quot;</span><span class="p">;</span> havana_transcript <span class="s2">&quot;OTTHUMT00000001365.1&quot;</span><span class="p">;</span>
chr1    HAVANA  exon    <span class="m">1693391</span> <span class="m">1693474</span> .   -   .   gene_id <span class="s2">&quot;ENSG00000008130.11&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENST00000341991.3&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;NADK&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;NADK-002&quot;</span><span class="p">;</span> exon_number <span class="m">3</span><span class="p">;</span>  exon_id <span class="s2">&quot;ENSE00003487616.1&quot;</span><span class="p">;</span>  level <span class="m">2</span><span class="p">;</span> tag <span class="s2">&quot;basic&quot;</span><span class="p">;</span> tag <span class="s2">&quot;CCDS&quot;</span><span class="p">;</span> ccdsid <span class="s2">&quot;CCDS30565.1&quot;</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000942.5&quot;</span><span class="p">;</span> havana_transcript <span class="s2">&quot;OTTHUMT00000002768.1&quot;</span><span class="p">;</span>
chr1    HAVANA  CDS <span class="m">1688280</span> <span class="m">1688321</span> .   -   <span class="m">0</span>   gene_id <span class="s2">&quot;ENSG00000008130.11&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENST00000497186.1&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;NADK&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;nonsense_mediated_decay&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;NADK-008&quot;</span><span class="p">;</span> exon_number <span class="m">2</span><span class="p">;</span>  exon_id <span class="s2">&quot;ENSE00001856899.1&quot;</span><span class="p">;</span>  level <span class="m">2</span><span class="p">;</span> tag <span class="s2">&quot;mRNA_start_NF&quot;</span><span class="p">;</span> tag <span class="s2">&quot;cds_start_NF&quot;</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000942.5&quot;</span><span class="p">;</span> havana_transcript <span class="s2">&quot;OTTHUMT00000002774.3&quot;</span><span class="p">;</span>
</code></pre></div>
<blockquote>
<p>The transcriptome has 9 columns. The first 8 are separated by tabs and look reasonable (chromosome, annotation source, feature type, start, end, score, strand, and phase), the last one is kind of hairy: it is made up of key-value pairs separated by semicolons, some fields are mandatory and others are optional, and the values are surrounded in double quotes. Thatâ€™s no way to live a decent life. (<em>text copied from the source</em>)</p>
</blockquote>
<p>let's get only the lines that have <code>gene</code> in the 3<sup>th</sup> column.</p>
<div class="highlight"><pre><span></span><code>$ awk -F <span class="s1">&#39;$3 == &quot;gene&quot;&#39;</span> transcriptome.gtf <span class="p">|</span> head <span class="p">|</span> less -S

chr1    HAVANA  gene    <span class="m">11869</span>   <span class="m">14412</span>   .   +   .   gene_id <span class="s2">&quot;ENSG00000223972.4&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000223972.4&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;DDX11L1&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;DDX11L1&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">14363</span>   <span class="m">29806</span>   .   -   .   gene_id <span class="s2">&quot;ENSG00000227232.4&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000227232.4&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;WASH7P&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;WASH7P&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000958.1&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">29554</span>   <span class="m">31109</span>   .   +   .   gene_id <span class="s2">&quot;ENSG00000243485.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000243485.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;MIR1302-11&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;MIR1302-11&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000959.2&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">34554</span>   <span class="m">36081</span>   .   -   .   gene_id <span class="s2">&quot;ENSG00000237613.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000237613.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;FAM138A&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;FAM138A&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000960.1&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">52473</span>   <span class="m">54936</span>   .   +   .   gene_id <span class="s2">&quot;ENSG00000268020.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000268020.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;OR4G4P&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;OR4G4P&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000185779.1&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">62948</span>   <span class="m">63887</span>   .   +   .   gene_id <span class="s2">&quot;ENSG00000240361.1&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000240361.1&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;OR4G11P&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;OR4G11P&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001095.2&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">69091</span>   <span class="m">70008</span>   .   +   .   gene_id <span class="s2">&quot;ENSG00000186092.4&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000186092.4&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;OR4F5&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;OR4F5&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001094.1&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">89295</span>   <span class="m">133566</span>  .   -   .   gene_id <span class="s2">&quot;ENSG00000238009.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000238009.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;RP11-34P13.7&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;RP11-34P13.7&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001096.2&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">89551</span>   <span class="m">91105</span>   .   -   .   gene_id <span class="s2">&quot;ENSG00000239945.1&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000239945.1&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;RP11-34P13.8&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;RP11-34P13.8&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001097.2&quot;</span><span class="p">;</span>
chr1    HAVANA  gene    <span class="m">131025</span>  <span class="m">134836</span>  .   +   .   gene_id <span class="s2">&quot;ENSG00000233750.3&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000233750.3&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;CICP27&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;CICP27&quot;</span><span class="p">;</span> level <span class="m">1</span><span class="p">;</span> tag <span class="s2">&quot;pseudo_consens&quot;</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001257.3&quot;</span><span class="p">;</span>
</code></pre></div>
<p>Perhaps filter a bit more and print the content of the 9<sup>th</sup> column in the file.</p>
<div class="highlight"><pre><span></span><code>$ awk -F <span class="s2">&quot;\t&quot;</span> <span class="s1">&#39;$3 == &quot;gene&quot; { print $9 }&#39;</span> transcriptome.gtf <span class="p">|</span> head <span class="p">|</span> less -S

gene_id <span class="s2">&quot;ENSG00000223972.4&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000223972.4&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;DDX11L1&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;DDX11L1&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000961.2&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000227232.4&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000227232.4&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;WASH7P&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;WASH7P&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000958.1&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000243485.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000243485.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;MIR1302-11&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;MIR1302-11&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000959.2&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000237613.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000237613.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;FAM138A&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;FAM138A&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000000960.1&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000268020.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000268020.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;OR4G4P&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;OR4G4P&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000185779.1&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000240361.1&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000240361.1&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;OR4G11P&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;OR4G11P&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001095.2&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000186092.4&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000186092.4&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;OR4F5&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;protein_coding&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;OR4F5&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001094.1&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000238009.2&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000238009.2&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;RP11-34P13.7&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;RP11-34P13.7&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001096.2&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000239945.1&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000239945.1&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;RP11-34P13.8&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;lincRNA&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;NOVEL&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;RP11-34P13.8&quot;</span><span class="p">;</span> level <span class="m">2</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001097.2&quot;</span><span class="p">;</span>
gene_id <span class="s2">&quot;ENSG00000233750.3&quot;</span><span class="p">;</span> transcript_id <span class="s2">&quot;ENSG00000233750.3&quot;</span><span class="p">;</span> gene_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> gene_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> gene_name <span class="s2">&quot;CICP27&quot;</span><span class="p">;</span> transcript_type <span class="s2">&quot;pseudogene&quot;</span><span class="p">;</span> transcript_status <span class="s2">&quot;KNOWN&quot;</span><span class="p">;</span> transcript_name <span class="s2">&quot;CICP27&quot;</span><span class="p">;</span> level <span class="m">1</span><span class="p">;</span> tag <span class="s2">&quot;pseudo_consens&quot;</span><span class="p">;</span> havana_gene <span class="s2">&quot;OTTHUMG00000001257.3&quot;</span><span class="p">;</span>
</code></pre></div>
<p>What about if we want just a specific piece from this information?
We can <code>|</code> the output from the first awk script in to a second one. Note that we will use different field separator <code>"; "</code>.</p>
<div class="highlight"><pre><span></span><code>$ awk -F <span class="s2">&quot;\t&quot;</span> <span class="s1">&#39;$3 == &quot;gene&quot; { print $9 }&#39;</span> transcriptome.gtf <span class="p">|</span> awk -F <span class="s2">&quot;; &quot;</span> <span class="s1">&#39;{ print $3 }&#39;</span> <span class="p">|</span> head

gene_type <span class="s2">&quot;pseudogene&quot;</span>
gene_type <span class="s2">&quot;pseudogene&quot;</span>
gene_type <span class="s2">&quot;lincRNA&quot;</span>
gene_type <span class="s2">&quot;lincRNA&quot;</span>
gene_type <span class="s2">&quot;pseudogene&quot;</span>
gene_type <span class="s2">&quot;pseudogene&quot;</span>
gene_type <span class="s2">&quot;protein_coding&quot;</span>
gene_type <span class="s2">&quot;lincRNA&quot;</span>
gene_type <span class="s2">&quot;lincRNA&quot;</span>
gene_type <span class="s2">&quot;pseudogene&quot;</span>
</code></pre></div>
<p>At this point I will suggest to follow the original tutorial:<br />
<a href="http://reasoniamhere.com/2013/09/16/awk-gtf-how-to-analyze-a-transcriptome-like-a-pro-part-1/">AWK GTF! How to Analyze a Transcriptome Like a Pro</a></p>
                
              
              
                


              
            </article>
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-footer-copyright">
        
        Made with
        <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
          Material for MkDocs
        </a>
      </div>
      <a href="https://uppmax.uu.se/" target="_blank" rel="noopener"><img align="right" alt="Uppsala" src="https://pmitev.github.io/to-awk-or-not/images/Uppsala.png" /> </a>
      <a href="https://snic.se/" target="_blank" rel="noopener"><img align="right" alt="SNIC" src="https://pmitev.github.io/to-awk-or-not/images/SNIC.png" /> </a>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../..", "features": [], "translations": {"clipboard.copy": "Copy to clipboard", "clipboard.copied": "Copied to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.placeholder": "Type to start searching", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.term.missing": "Missing", "select.version.title": "Select version"}, "search": "../../assets/javascripts/workers/search.409db549.min.js", "version": null}</script>
    
    
      <script src="../../assets/javascripts/bundle.756773cc.min.js"></script>
      
        <script src="../../javascripts/config.js"></script>
      
        <script src="https://polyfill.io/v3/polyfill.min.js?features=es6"></script>
      
        <script src="https://cdn.jsdelivr.net/npm/mathjax@3/es5/tex-mml-chtml.js"></script>
      
    
  </body>
</html>